src/server/project.ts

```diff
@@ -1114,6 +1114,7 @@ namespace ts.server {
                 // If typing files changed, then only schedule project update
                 this.typingFiles = typingFiles;
                 // Invalidate files with unresolved imports
+                this.resolutionCache.invalidateTypingsFiles();

```

Comment:
It seems like it would be possible to tell the resolution cache exactly which typings files were added/removed. But I’m not sure it’s worth it, because `resolveTypeReferenceDirective` already short-circuits on resolutions that were previously successful.